### ISSUE TITLE
iput  -b -r not always calling acPostProcForPut when post put policy is on

### ISSUE THREAD

**Issue by tedgin** [irods/irods#2841]
I have the following rules set.

```
acBulkPutPostProcPolicy { msiSetBulkPutPostProcPolicy("on"); }
acPostProcForPut { writeLine("serverLog", "acPostProcForPut called for $objPath"); }
```

When I perform a bulk put on a directory with five files in it, the `acPostProcForPut` PEP is only called for two of the five files.

**Comment by tedgin** [comment]
Would it be possible to get this fixed in the 4.1.5 release? We have customers who rely on this feature to upload large data sets.

**Comment by trel** [comment]
i'm going to try and reproduce this.

**Comment by trel** [comment]
This appears on 4-1-stable only when the files are smaller than ~4MB.

The following new test traps the bug.  It starts with files smaller than the threshold and ends with files that already work as expected.

```
    @unittest.skip('waiting for fix')
    def test_iput_bulk_check_acpostprocforput__2841(self):
        # prepare test directory
        number_of_files = 5
        dirname = self.admin.local_session_dir + '/files'
        corefile = lib.get_core_re_dir() + "/core.re"
        # files less than 4200000 were failing to trigger the writeLine
        for filesize in range(5000, 6000000, 500000):
            files = lib.make_large_local_tmp_dir(dirname, number_of_files, filesize)
            # manipulate core.re and check the server log
            with lib.file_backed_up(corefile):
                initial_size_of_server_log = lib.get_log_size('server')
                rules_to_prepend = '''
acBulkPutPostProcPolicy { msiSetBulkPutPostProcPolicy("on"); }
acPostProcForPut { writeLine("serverLog", "acPostProcForPut called for $objPath"); }
'''
                time.sleep(1)  # remove once file hash fix is committed #2279
                lib.prepend_string_to_file(rules_to_prepend, corefile)
                time.sleep(1)  # remove once file hash fix is committed #2279
                self.admin.assert_icommand(['iput', '-rb', dirname])
                assert number_of_files == lib.count_occurrences_of_string_in_log(
                    'server', 'writeLine: inString = acPostProcForPut called for', start_index=initial_size_of_server_log)
                os.unlink(files)
```
